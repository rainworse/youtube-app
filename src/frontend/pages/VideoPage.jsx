import { useParams } from 'react-router-dom';
import LocalStorageDB from '../../backend/LocalStorageDB';
import VideoPlayer from '../components/VideoPlayer';
import '../styles/VideoPage.css';
import CommentSection from '../components/CommentSection';
import ConfigHelper from '../../backend/ConfigHelper';
import VideoContainer from '../components/VideoContainer';
import APIFetcher from '../../backend/APIFetcher';

const VideoPage = () => {
  const params = useParams();
  const suggestionCount = ConfigHelper.getValue('video-page-suggestion-count');

  return (
    <div className="video-page">
      <div className="video-section">
        <div className="video-player">
          <VideoPlayer src={LocalStorageDB.getVideoURL(params.id)} />
        </div>

        <div className="video-comment-section">
          <CommentSection videoID={params.id} />
        </div>
      </div>

      <div className="video-suggestion-sidebar">
        <VideoContainer
          fetchMethod={APIFetcher.fetchSuggestedVideoSnippets}
          fetchParams={[suggestionCount, params.id]}
        />
      </div>
    </div>
  );
};

export default VideoPage;
