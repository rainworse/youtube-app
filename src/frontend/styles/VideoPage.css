.video-page {
  display: flex;
  width: 100%;
  height: 100%;
}

.video-section {
  margin-left: 10%;
  margin-right: auto;
  display: flex;
  flex-direction: column;
  width: 53%;
}

.video-player {
  width: 100%;
}

#iframe-video-player {
  border-radius: 15px;
  border: none;
}

.video-suggestion-sidebar {
  width: 25%;
  flex-shrink: 1;
  display: flex;
  flex-direction: column;
  gap: 20px;
  margin-left: 5%;
  margin-right: 5%;
}

.video-suggestion-sidebar > .video-card {
  position: relative;
}

.video-comment-section {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  width: auto;
  /* height: 500px; */
}

.add-comment-form {
  display: flex;
  margin-top: 10px;
  margin-bottom: 20px;
  max-width: 100%;
  align-items: center;
}

#add-comment-field {
  resize: none;
  flex: 1;
  box-sizing: border-box;
  max-width: 100%;
  height: 40px;
  font-size: 16px;
  margin-right: -40px;
  padding: 10px;
  border: none;
  border-radius: 15px;
  font-family: 'Tinos', serif;
}

#post-comment {
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  background-color: white;
  font-family: 'Tinos', serif;
  font-weight: bold;
}
