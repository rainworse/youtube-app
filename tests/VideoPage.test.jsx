import { fireEvent, render, waitFor } from '@testing-library/react';
import { expect, vi } from 'vitest';
import VideoPage from '../src/frontend/pages/VideoPage';
import UserContext from '../src/frontend/UserContext';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

vi.mock('../src/backend/APIFetcher');
vi.mock('../src/backend/LocalStorageDB');

describe('VideoPage', () => {
  it('renders video page', async () => {
    const videoPage = render(
      <UserContext.Provider value={{ currentUser: 'user' }}>
        <MemoryRouter>
          <VideoPage />
        </MemoryRouter>
      </UserContext.Provider>
    );

    await waitFor(() => videoPage.getByText('test video1'));
    await waitFor(() => videoPage.getByText('john pork'));
    await waitFor(() => videoPage.getByText('local man'));

    expect(videoPage.container).toMatchSnapshot();
  });

  // it('add comment to video page', async () => {
  //   // const user = userEvent.setup()

  //   const videoPage = render(
  //     <UserContext.Provider value={{ currentUser: 'testUserComment' }}>
  //       <MemoryRouter>
  //         <VideoPage />
  //       </MemoryRouter>
  //     </UserContext.Provider>
  //   );

  //   await waitFor(() => videoPage.getByText('test video1'));
  //   await waitFor(() => videoPage.getByText('john pork'));
  //   await waitFor(() => videoPage.getByText('local man'));

  //   const commentField = videoPage.getByPlaceholderText('Add a comment...');
  //   const postButton = videoPage.getByText('Post');

  //   fireEvent.change(commentField, {
  //     target: { value: 'a newly posted comment' },
  //   });
  //   fireEvent.click(postButton);

  //   // await user.type(commentField, 'a newly posted comment');
  //   // await user.click(postButton);

  //   await waitFor(() => videoPage.getByText('a newly posted comment'));
  //   expect(videoPage.container).toMatchSnapshot();
  // });
});
