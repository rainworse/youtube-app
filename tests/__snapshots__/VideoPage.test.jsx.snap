// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VideoPage > renders video page 1`] = `
<div>
  <div
    class="video-page"
  >
    <div
      class="video-section"
    >
      <div
        class="video-player"
      >
        <iframe
          height="303.1578947368421"
          id="iframe-video-player"
          src="fake/video/url"
          width="538.9473684210526"
        />
      </div>
      <div
        class="video-comment-section"
      >
        <form
          class="add-comment-form"
        >
          <textarea
            id="add-comment-field"
            name="add-comment"
            placeholder="Add a comment..."
          />
          <input
            id="post-comment"
            type="submit"
            value="Post"
          />
        </form>
        <div
          class="video-comment"
        >
          <img
            alt="image"
            class="video-comment-author-image"
            src="fake/urllocal"
          />
          <div
            class="video-comment-content"
          >
            <div
              class="video-comment-author-name"
            >
              local man
            </div>
            <div
              class="video-comment-text"
            >
              local comment
            </div>
          </div>
        </div>
        <div
          class="video-comment"
        >
          <img
            alt="image"
            class="video-comment-author-image"
            src="fake/url1"
          />
          <div
            class="video-comment-content"
          >
            <div
              class="video-comment-author-name"
            >
              john pork
            </div>
            <div
              class="video-comment-text"
            >
              sample comment
            </div>
          </div>
        </div>
        <div
          class="video-comment"
        >
          <img
            alt="image"
            class="video-comment-author-image"
            src="fake/url2"
          />
          <div
            class="video-comment-content"
          >
            <div
              class="video-comment-author-name"
            >
              richard d james
            </div>
            <div
              class="video-comment-text"
            >
              another comment
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="video-suggestion-sidebar"
    >
      <div
        class="video-card"
      >
        <a
          class="video-card-thumbnail-anchor"
          href="/video/001"
        >
          <img
            alt="test video1"
            class="video-card-thumbnail"
            src="non/existent/location1"
          />
          <div
            class="video-card-text"
          >
            <div
              class="video-card-title"
            >
              test video1
            </div>
            <a
              class="video-card-channel-anchor"
              href="/channel/channel001"
            >
              <div
                class="video-card-channel"
              >
                by 
                test channel1
              </div>
            </a>
          </div>
        </a>
      </div>
      <div
        class="video-card"
      >
        <a
          class="video-card-thumbnail-anchor"
          href="/video/002"
        >
          <img
            alt="test video2"
            class="video-card-thumbnail"
            src="non/existent/location2"
          />
          <div
            class="video-card-text"
          >
            <div
              class="video-card-title"
            >
              test video2
            </div>
            <a
              class="video-card-channel-anchor"
              href="/channel/channel002"
            >
              <div
                class="video-card-channel"
              >
                by 
                test channel2
              </div>
            </a>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
